File: AD-HOC/1553/main.js
function getFrequencies(p) {
    let frequencies = {};
    for (let question of p) {
        frequencies[question] = question in frequencies ? frequencies[question] + 1 : 1;
    }
    return Object.values(frequencies);
}

function getQuestions(frequencies, k) {
    let questions = 0;
    for (let frequency of frequencies) {
        questions += frequency >= k;
    }
    return questions;
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
let [n, k] = lines.shift().split(' ').map(Number);
while (n > 0 && k > 0) {
    let p = lines.shift().split(' ').map(Number);
    let frequencies = getFrequencies(p);
    console.log(getQuestions(frequencies, k));
    [n, k] = lines.shift().split(' ').map(Number);
}

File: AD-HOC/1553/main.rb
def get_frequencies(p)
  frequencies = {}
  p.each do |question|
    frequencies[question] = frequencies.include?(question) ? frequencies[question] + 1 : 1
  end
  frequencies.values
end

def get_questions(frequencies, k)
  questions = 0
  frequencies.each do |frequency|
    questions += 1 if frequency >= k
  end
  questions
end

n, k = gets.split.map(&:to_i)
while n.nonzero? & k.nonzero?
  p = gets.split.map(&:to_i)
  frequencies = get_frequencies(p)
  puts get_questions(frequencies, k)
  n, k = gets.split.map(&:to_i)
end
